src/com/lundekhan/stokk/StokkRoute.kt

```diff
@@ -54,28 +55,28 @@ fun Route.stokkRoute(): Route = route("/stokk") {
     get { call.respondHtmlShell { section { stokkForm() } } }
     post {
         val params = call.receiveParameters()
-        val years = params["year"]?.toInt() ?: 0
-        val startAmount = params["startAmount"]?.toDouble() ?: 0.0
-        val interest = params["interest"]?.toDouble() ?: 0.0
-        val monthlyDeposit = params["monthlyDeposits"]?.toInt() ?: 0
-        val compoundedData = StokkHelper.compoundInterest(
+        val years = params["year"]?.toIntOrNull() ?: 0
+        val startAmount = params["startAmount"]?.toDoubleOrNull() ?: 0.0
+        val interest = params["interest"]?.toDoubleOrNull() ?: 0.0
+        val monthlyDeposit = params["monthlyDeposits"]?.toIntOrNull() ?: 0
+        val compoundedData = StokkHelper.compoundInterestMonthByMonth(
             years, startAmount,
             interest / 100 + 1.0, monthlyDeposit
         )
 
         call.respondHtmlShell {
             section {
                 stokkForm(
-                    params["year"]!!,
-                    params["startAmount"]!!,
-                    params["interest"]!!,
-                    params["monthlyDeposits"]!!
+                        years.toString(),
+                        startAmount.toString(),
+                        interest.toString(),
+                        monthlyDeposit.toString()
                 )
             }
             section {
                 aside {
                     h3 { +"Result: " }
-                    +compoundedData.roundToInt().toString()
+                    +compoundedData.last().roundToInt().toString()
```

Comment:
If you wanted to do the `lastOrNull` I think the cleanest approach is `.lastOrNull()?.roundToInt()?.toString() ?: "0"`